symengine/pow.cpp

```diff
@@ -108,8 +108,12 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         }
     }
 
-    if (eq(*a, *one))
+    if (eq(*a, *one)) {
+        if (eq(*b, *Inf) || eq(*b, *NegInf))
```

Comment:
I don't really like special casing for just `oo`. Instead of checking for `oo, -oo, nan, zoo` etc, can you check that `b` is a number and if so call `pownum`?

Reply:
on second thought this is handled below. So, you can change the original `if` condition to be in pseudo-code, `if (a == 1 and b is not a number)`

Reply:
If I'm understanding you correctly, `if (eq(*a, *one) and !is_a_Number(*b)) return Nan;` is the change you're asking me to do. However, I'm getting failed tests after making the changes.

Reply:
`if (eq(*a, *one) and !is_a_Number(*b)) return one;`